File: GamePi Angular/main/src/app/Models/Decision.ts
import { User } from './User'; 
import { Scenario } from './Scenario'; 

export class Decision {
    id!: number; 
    action!: Decision.ActionType; 
    date!: Date; 
    profitOrLoss?: number; 
  
    
    user?: User; 
    scenario?: Scenario; 
  
    constructor(init?: Partial<Decision>) {
      Object.assign(this, init);
    }
  }
  
  
  export namespace Decision {
    export enum ActionType {
      BUY = 'BUY',
      SELL = 'SELL',
      HOLD = 'HOLD',
    }
  }
  

File: GamePi Spring/src/main/java/tn/esprit/gamepi/entities/Decision.java
package tn.esprit.gamepi.entities;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.io.Serializable;
import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@RequiredArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
public class Decision implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    @NonNull
    @Enumerated(EnumType.STRING)
    ActionType action;

    @NonNull
    LocalDate date;

    double profitOrLoss;

    @ManyToOne
    @JoinColumn(name = "user_id")
    User user;

    @ManyToOne
    @JoinColumn(name = "scenario_id")
    Scenario scenario;
}
